onnx/defs/tensor/defs.cc

```diff
@@ -242,37 +242,22 @@ will be (2, 1, 3).
         {"tensor(float16)", "tensor(float)", "tensor(double)"},
         "Constrain input and output types to float tensors.")
     .ShapeInferenceFunction([](InferenceContext& ctx) {
-      if (ctx.getNumInputTypes() != 1) {
-        return;
-      }
-      std::vector<int64_t> perm;
-      {
-        auto perm_attr = ctx.getAttribute("perm");
-        if (perm_attr) {
-          for (auto dim : perm_attr->ints()) {
-            perm.push_back(dim);
-          }
-        } else {
-          int ndims = ctx.getInputType(0)->shape().dim_size();
-          for (int i = 0; i < ndims; i++) {
-            perm.push_back(ndims - i - 1);
-          }
+        if (ctx.getNumInputTypes() != 1 || !ctx.getInputType(0)) {
```

Comment:
should we just enforce that it has only 1 input?

Reply:
seems like more the role of the checker